src/openrct2/ride/Ride.cpp

```diff
@@ -4400,7 +4400,7 @@ static Vehicle* vehicle_create_car(
         // loc_6DDCA4:
         vehicle->TrackSubposition = VEHICLE_TRACK_SUBPOSITION_0;
         int32_t direction = tileElement->GetDirection();
-        auto chosenLoc = CoordsXYZ{ CoordsXY{ carPosition } + word_9A3AB4[direction], tileElement->GetBaseZ() };
+        auto chosenLoc = carPosition + CoordsXYZ{ word_9A3AB4[direction], 0 };
```

Comment:
This is what I meant @Gymnasiast . The call stack is
```
vehicle_create_car
vehicle_create_train
vehicle_create_trains
```
Though the z position was being sent to these functions, it was not being used and it was initialized from the `tileElement` instead. I'm now using the `z` as sent. I'll annotate the other places